src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -695,10 +695,17 @@ Task<bool> IsIntentListenerRegisteredAsync(AppMetadata appMetadata)
 
     private Task RemoveModuleAsync(IModuleInstance instance)
     {
-        var fdc3InstanceId = GetFdc3InstanceId(instance);
-        if (!_runningModules.TryRemove(new(fdc3InstanceId), out _))
+        try
+        {
+            var fdc3InstanceId = GetFdc3InstanceId(instance);
+            if (!_runningModules.TryRemove(new(fdc3InstanceId), out _))
+            {
+                _logger.LogError($"Could not remove the closed window with instanceId: {fdc3InstanceId}.");
+            }
+        }
+        catch (Fdc3DesktopAgentException exception)
```

Comment:
This exception is catched because if that is not a fdc3 enabled app aka it has no fdc3instanceId then we shouldn't care if the GetFdc3InstanceId(IModuleInstance) throws an error as the fdc3InstanceId is missing. If you start the messagerouter examples: chart and grid then you experiment this issue with the current shell on close if you set the exception settings to catch all.

Reply:
If it's not an FDC3 app then why are we even here in Fdc3DesktopAgent?

Reply:
So once we enable FDC3 we can no longer run non-FDC3 applications? That sounds bad, and the solution won't be an empty `catch` block.
For now I'm OK with this change because it is not related to your main change, but there should be a follow-up to fix this properly.